tests/test_custom_models.py

```diff
@@ -852,6 +856,46 @@ def forward(self, X):
         return X
 
 
+class ModelConv2D1x1(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.conv2d = nn.Conv2d(1, 10, kernel_size=(1, 1), padding=0)
+        self.relu = nn.ReLU()
+        self.flat = nn.Flatten()
+        self.lin0 = nn.Linear(10 * 5 * 5, 2)
+        self.sm = nn.LogSoftmax(dim=-1)   
+        self.dtype = torch.float          
+
+    def forward(self, X):
+        X = X.to(self.dtype)
+        X = X.reshape(-1, 1, 5, 5)  
+        X = self.conv2d(X)
+        X = self.relu(X)
+        X = self.flat(X)
+        X = self.lin0(X)
+        X = self.sm(X)
+        return X
+
+
+class ModelConv1D(nn.Module):
```

Comment:
Let's use `Conv1d` rather than `Conv1D` to avoid confusion with the already confusing `Conv1D` implementation from transformers.